crates/nu-protocol/src/value/record.rs

```diff
@@ -3,13 +3,68 @@ use std::{iter::FusedIterator, ops::RangeBounds};
 
 use crate::{ShellError, Span, Value};
 
+use nu_utils::IgnoreCaseExt;
 use serde::{Deserialize, Serialize, de::Visitor, ser::SerializeMap};
 
 #[derive(Debug, Clone, Default)]
 pub struct Record {
     inner: Vec<(String, Value)>,
 }
 
+pub struct CasedRecord<R> {
+    record: R,
+    insensitive: bool,
+}
```

Comment:
I think we should strieve for docs in `nu-protocol`. Could you please explain for what `CasedRecord` is used. From the name and fields I have a vague idea but its usages aren't totally clear to me. If you could explain that in the docs, that would be nice. So why and when do we use that struct. And maybe even why `CasedRecord` even exists next to `Record`.

The methods are fine, I don't need docs for them, but this struct confuses me a bit.